crates/bevy_dev_tools/src/fps_overlay.rs

```diff
@@ -86,13 +170,75 @@ fn update_text(diagnostic: Res<DiagnosticsStore>, mut query: Query<&mut Text, Wi
     }
 }
 
+fn update_frame_time_values(
+    mut frame_time_graph_materials: ResMut<Assets<FrametimeGraphMaterial>>,
+    diagnostics_store: Res<DiagnosticsStore>,
+) {
+    let Some(frame_time) = diagnostics_store.get(&FrameTimeDiagnosticsPlugin::FRAME_TIME) else {
+        return;
+    };
+    let frame_times = frame_time
+        .values()
+        // convert to millis
+        .map(|x| *x as f32 / 1000.0)
+        .collect::<Vec<_>>();
+    for (_, material) in frame_time_graph_materials.iter_mut() {
+        material.values = frame_times.clone();
+    }
+}
+
 fn customize_text(
     overlay_config: Res<FpsOverlayConfig>,
     mut query: Query<&mut Text, With<FpsText>>,
+    mut graph_style: Query<&mut Style, With<FrameTimeGraph>>,
 ) {
     for mut text in &mut query {
         for section in text.sections.iter_mut() {
             section.style = overlay_config.text_config.clone();
         }
     }
+    if overlay_config.frame_time_graph_config.enabled {
+        // Scale the frame time graph based on
+        let font_size = overlay_config.text_config.font_size;
+        let mut graph_style = graph_style.single_mut();
+        graph_style.width = Val::Px(font_size * FRAME_TIME_GRAPH_WIDTH_SCALE);
+        graph_style.height = Val::Px(font_size * FRAME_TIME_GRAPH_HEIGHT_SCALE);
+    }
+}
+
+#[derive(Debug, Clone, Copy, ShaderType)]
+struct FrameTimeGraphConfigUniform {
+    dt_min: f32,
+    dt_max: f32,
+    dt_min_log2: f32,
+    dt_max_log2: f32,
+    proportional_width: u32,
+}
+impl FrameTimeGraphConfigUniform {
+    fn new(dt_min: f32, dt_max: f32, proportional_width: bool) -> Self {
+        Self {
+            dt_min,
+            dt_max,
+            dt_min_log2: dt_min.log2(),
+            dt_max_log2: dt_max.log2(),
+            proportional_width: u32::from(proportional_width),
+        }
+    }
 }
+
+#[derive(AsBindGroup, Asset, TypePath, Debug, Clone)]
+struct FrametimeGraphMaterial {
```

Comment:
Is there a way to have this be public and allow easy reuse for consumers?

In my project I would want to place the frame time graph in a certain location within my UI, and if it was just an accessible UiMaterial that might work out.

Reply:
I'm also happy with having this in a future PR if out of scope.

The only thing needed afaik is to:

1. Make `FrametimeGraphMaterial` public
2. Don't add the UI node (in `setup` in this PR) unconditonally

Spawning the UI node is then up to the user.

Reply:
The material on it's own doesn't really make sense without the system that updates the values. 

The point of the devtool is to make it as simple as possible to just drop in. Forcing users to spawn ui nodes for this instead of just adding the plugin doesn't seem like a good compromise to me. I don't mind making all the structs/systems pub if people want to use it separately though.

Reply:
I made everything pub including the system so it's useable without the plugin if you are willing to add/spawn everything manually.

Reply:
What I had in mind was a way to add the plugin where all systems (including the one updating materials) is added via the plugin like you do now so the only thing a user has to do is to add the material in their own UI node tree:

```rust
                let font_size = overlay_config.text_config.font_size;
                c.spawn((
                    MaterialNodeBundle {
                        style: Style {
                            width: Val::Px(font_size * FRAME_TIME_GRAPH_WIDTH_SCALE),
                            height: Val::Px(font_size * FRAME_TIME_GRAPH_HEIGHT_SCALE),
                            ..default()
                        },
                        material: frame_time_graph_materials.add(FrametimeGraphMaterial {
                            values: vec![],
                            config: FrameTimeGraphConfigUniform::new(
                                // we want an upper limit that is above the target otherwise the bars will disappear
                                1. / (overlay_config.frame_time_graph_config.max_fps * 1.2),
                                1. / overlay_config.frame_time_graph_config.min_fps,
                                true,
                            ),
                        }),
                        ..default()
                    },
                    FrameTimeGraph,
                ));
```

^-- basically this should be what I'd add to my own UI container somewhere and the plugin you are making would update it since you are adding

```rust
    mut frame_time_graph_materials: ResMut<Assets<FrametimeGraphMaterial>>,
```

which updates materials no matter who spawned them.

I think that should work?

Reply:
Right, I guess I could extract it to it's own plugin and then just spawn it in the `FpsOverlay` plugin